openlibrary/plugins/openlibrary/js/carousels.js

```diff
@@ -1,6 +1,6 @@
 // used in templates/covers/add.html
 const Carousel = {
```

Comment:
This should have some jsdoc function documentation. It's not clear what the types of the different and many parameters are and what purpose they serve. It's also not clear it has side effects! All should be documented.
e.g.
```
/**
 * function that does X
 * @param {string} selector for doing x
 * @param {string} a for doing y
```

Reply:
What is the right way to jsdoc comment an object such that annotations can be left on a e.g. typedef? i.e. how do I add a doc for "getItems"
```
    /**                                                                                                                                                                                                                             
     * function Carousel.add instantiates a new slick carousel and optionally                                                                                                                                                       
     * loadMore params for lazy-loading more pages of results on the fly                                                                                                                                                            
     * @param {string} selector to bind carousel                                                                                                                                                                                    
     * @param {string} a - f are number of items to render at different mobile breakpoints                                                                                                                                          
     * @param {{                                                                                                                                                                                                                    
     *   url:string endpoint for fetching additional results                                                                                                                                                                        
     *   getItems:function which extracts item values out of the API response                                                                                                                                                       
     *   addItem:function which consumes and item and returns html to add as a new slide                                                                                                                                            
     *   limit:int number of slides to add when end is hit                                                                                                                                                                          
     *   pageMode:string which specifies if API uses 'page' or 'offset' as its mechanism                                                                                                                                            
     * }} loadMore is a dict of options for lazy-loading and includes:                                                                                                                                                              
     */
```